packages/features/ee/billing/credit-service.ts

```diff
@@ -30,41 +30,45 @@ export class CreditService {
     bookingUid?: string;
     smsSid?: string;
   }) {
-    let teamToCharge = credits === 0 && teamId ? teamId : null;
+    let teamIdToCharge = credits === 0 && teamId ? teamId : undefined;
```

Comment:
Many things are happening in this function, and I suggest we use $transaction to make all the operations atomic. What do you think?